projects/components/src/table/cells/table-cell-csv-generator-base.ts

```diff
@@ -0,0 +1,13 @@
+import { Dictionary } from '@hypertrace/common';
+import { InjectionToken } from '@angular/core';
+
+export const TABLE_CELL_CSV_GENERATORS = new InjectionToken<unknown[][]>('TABLE_CELL_CSV_GENERATORS');
+
+export interface TableCellCsvGeneratorBase<TCellData, TRowData = unknown> {
+  readonly type: CSV_GENERATOR_TYPE;
+  readonly cellType: string;
+
+  generateSafeCsv(cellData?: TCellData, rowData?: TRowData): string | Dictionary<string> | undefined;
```

Comment:
When would we produce a dictionary?

Reply:
if the column is an object like `{ key1: "value", key2: "value" }`, returning a dictionary allows us to create separate entries in the CSV for these.

Reply:
interesting... but what happens if one row to the next doesn't have consistent keys?

Reply:
usually that does not happen. all objects corresponding to a column have the same structure. in the off chance there are optional keys, the generator should account for it.

Reply:
I don't think we have this guarantee, and we can't have inconsistent numbers of values being output for rows. can come back to this one separately, but avoid using it until we resolve this.